src/conn/conn_log.c

```diff
@@ -78,6 +78,49 @@ __logmgr_force_archive(WT_SESSION_IMPL *session, uint32_t lognum)
     return (0);
 }
 
+/*
+ * __wt_logmgr_compat_version --
+ *     Set up the compatibility versions in the log manager. This is split out because it is called
+ *     much earlier than log subsystem creation on startup so that we can verify the system state in
+ *     files before modifying files.
+ */
+void
+__wt_logmgr_compat_version(WT_SESSION_IMPL *session)
+{
+    WT_CONNECTION_IMPL *conn;
+
+    conn = S2C(session);
+    /*
+     * Set up the maximum and minimum log version required if needed.
+     */
+    if (conn->req_max_major != WT_CONN_COMPAT_NONE) {
+        if (conn->req_max_major == WT_LOG_V5_MAJOR)
+            conn->log_req_max = WT_LOG_VERSION;
+        else if (conn->req_max_major == WT_LOG_V4_MAJOR)
+            if (conn->req_max_minor == WT_LOG_V4_MINOR)
+                conn->log_req_max = 4;
+            else if (conn->req_max_minor > WT_LOG_V2_MINOR)
+                conn->log_req_max = 3;
+            else
+                conn->log_req_max = 2;
+        else
+            conn->log_req_max = 1;
+    }
+    if (conn->req_min_major != WT_CONN_COMPAT_NONE) {
```

Comment:
Since this calculation is the same as the one above (except for `conn->req_min_major` as input, and `conn->log_req_min` as output), maybe it makes sense to refactor it into a function?

Reply:
Yes, I can do that. This comment got a bit lost with the other issue we discussed.

Reply:
Fixed.